extras/spiffs/mkspiffs/mkspiffs.c

```diff
@@ -187,12 +187,27 @@ static bool process_directory(const char *direcotry)
                 !strcmp(ep->d_name, "..")) {
                 continue;
             }
+            if(ep->d_type == DT_DIR) {
+                char *new_dir_name = malloc(strlen(direcotry) + strlen(ep->d_name) + 2);
+                sprintf(new_dir_name, "%s/%s", direcotry, ep->d_name);
+                process_directory(new_dir_name);
+                free(new_dir_name);
+                continue;
+            }
             if (ep->d_type != DT_REG) {
                 continue;  // not a regular file
             }
             sprintf(path, "%s/%s", direcotry, ep->d_name);
-            printf("Processing file %s\n", path);
-            if (!process_file(path, ep->d_name)) {
+            filename = path;
+            file_ptr = path;
+            while(*file_ptr != 0 && *file_ptr != '/') {
+                file_ptr ++;
+            }
+            if(*file_ptr != 0) {
+                filename = file_ptr + 1;
+            }
```

Comment:
As far as I understand this block of code sets `filename` to point past the fist occurrence  of '/'.
I may suggest alternative implementation:
```c
filename = strchr(path, '/');
filename = filename ? &filename[1] : path;
```